homeassistant/components/climate/__init__.py

```diff
@@ -96,7 +63,7 @@
         vol.Inclusive(ATTR_TARGET_TEMP_HIGH, 'temperature'): vol.Coerce(float),
         vol.Inclusive(ATTR_TARGET_TEMP_LOW, 'temperature'): vol.Coerce(float),
         vol.Optional(ATTR_ENTITY_ID): cv.comp_entity_ids,
-        vol.Optional(ATTR_OPERATION_MODE): cv.string,
+        vol.Optional(ATTR_OPERATION_MODE): vol.In(OPERATION_MODES),
```

Comment:
I think we should remove setting operation from the temperature service.

The reason we added it was to allow scenes to set both temperature and operation. We don't need this anymore after the change of reproduce state implementation, where each component defines how the state attributes should be reproduced using services.

Reply:
Well, I agree, but maybe you want cool and set a new temperature in one call?

Reply:
I think that we should either collapse all services in 1, or we don't .